app/controllers/tree_controller.rb

```diff
@@ -5,7 +5,19 @@ class TreeController < ApplicationController
   before_action :check_privileges
 
   def automate_entrypoint
-    json = fetch_tree(TreeBuilderAutomateEntrypoint, :automate_entrypoint_tree, params[:id])
+    json = fetch_tree(TreeBuilderAutomateEntrypoint, :automate_entrypoint_tree, params[:id]) do |tree|
+      if params[:fqname].present?
+        MiqAeInstance.get_homonymic_across_domains_without_domain_prefix(current_user, params[:fqname], true).each do |instance|
+          namespace, klass, instance, _ = MiqAeEngine::MiqAePath.split(instance.fqname)
+
+          open_nodes = namespace.split('/').map { |ns| MiqAeNamespace.find_by(:name => ns) }
+          open_nodes << MiqAeClass.find_by(:name => klass, :namespace_id => open_nodes.last.id)
+          open_nodes << MiqAeInstance.find_by(:name => instance, :class_id => open_nodes.last.id)
+
+          open_nodes.each { |node| tree.open_node(TreeNode.new(node, {}, {}).key) }
```

Comment:
@skateman It can happen that the automate methods are missing, and so for some dialogs, I'm getting `nil` at the end of `open_nodes` resulting in a exception.

Reply:
@romanblanco how it can happen? I need a little more to be able to work with it.

Reply:
@skateman testing database from h-kataria already has such records.

If you want to reproduce the issue on your own, I believe removing automate methods, already assigned to a dynamic field should do the trick

Reply:
Isn't this something that should be prevented on the activerecord level? I mean dropping stuff from elsewhere that we depend on?

Reply:
@skateman makes sense, not sure if this is the case as well
@pkomanek might know more

Reply:
Feel free to make this depend on a core PR adding foreign keys :trollface: ;)

Reply:
@lpichler cc

Reply:
@romanblanco I fixed the issue, however, instead of hacking around in the UI, this should be solved in the backend.